runhouse/main.py

```diff
@@ -679,6 +350,182 @@ def cluster_list(
+#     cluster_or_local = rh.here
+#     load_from_den = False
+#
+#     if cluster_or_local == "file" and not cluster_name:
+#         # If running outside the cluster must specify a cluster name
+#         console.print("Missing argument `cluster_name`.")
+#         return
+#
+#     elif cluster_name:
+#
+#         try:
+#             current_cluster = cluster(name=cluster_name)
+#         except ValueError:
+#             console.print("Failed to load status for cluster.")
+#             return
+#
+#         if not current_cluster.is_up():
+#             print_bring_cluster_up_msg(
+#                 cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+#             )
+#             load_from_den = True
+#
+#         try:
+#             if current_cluster._http_client:
+#                 current_cluster._http_client.check_server()
+#         except requests.exceptions.ConnectionError:
+#             print_bring_cluster_up_msg(
+#                 cluster_name=cluster_name, msg_prefix="Can't get live cluster logs"
+#             )
+#             load_from_den = True
+#
+#     elif not cluster_or_local:
+#         # todo: make func as the print_bring_cluster_up_msg
+#         console.print(
+#             "Can't get live cluster logs: \N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}. "
+#             "Start it with `runhouse restart` or specify a remote "
+#             "cluster to poll with `runhouse status <cluster_name>`."
+#         )
+#         load_from_den = True
+#
+#     else:
+#         # we are inside the cluster
+#         current_cluster = cluster_or_local  # cluster_or_local = rh.here
+#
+#     try:
+#         if load_from_den:
+#             console.print("Getting the last chunk of logs saved in Den.")
+#             get_logs_den_resp = get_saved_logs_from_den(current_cluster.rns_address)
+#             if get_logs_den_resp.status_code != 200:
+#                 console.print("Failed to get logs from den.")
+#                 return
+#             logs = "\n".join(get_logs_den_resp.json().get("data").get("logs_text"))
+#         else:
+#             logs = current_cluster.logs(tail)
+#
+#     except requests.exceptions.ConnectionError:
+#         console.print("Can't get live cluster logs, ")
+#         console.print(
+#             "\N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}"
+#         )
+#         return
+#
+#     console.print(f"[reset][italic cyan]{logs}")
+
+
+@cluster_app.command("up")
+def cluster_up(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to bring up.",
+    )
+):
+    """Bring up the cluster if it is not up. No-op if cluster is already up.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+        ``$ runhouse cluster up /sashab/rh-basic-cpu``
+
+    """
+    try:
+        current_cluster = rh.cluster(name=cluster_name)
```

Comment:
i think for this command (+ others where relevant) we should handle this being run on a cluster more gracefully - in this case that would mean not doing anything but printing a relevant message